internal/controller/clientprofile_controller.go

```diff
@@ -130,6 +131,10 @@ func (r *ClientProfileReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 func (r *ClientProfileReconcile) reconcile() error {
 	if err := r.loadAndValidate(); err != nil {
+		if k8serrors.IsNotFound(err) && r.clientProfile.UID == "" {
```

Comment:
@leelavg wouldn't it make more sense to investigate the error object to see the target of the not found error?